local_pathfinding/node_navigate.py

```diff
@@ -176,7 +178,22 @@ def get_desired_heading(self) -> float:
         )
 
         # TODO: create function to compute the heading from current position to next local waypoint
-        return 0.0
+        # Find index of self.gps in current waypoints
+        current_waypoint = self.gps.lat_lon
+        next_waypoint = None
+        waypoints = self.local_path.waypoints
+        for i in range(len(waypoints)):
+            if waypoints[i] == current_waypoint and i != len(waypoints) - 1:
```

Comment:
I think this should be `i < len(waypoints) - 1`. 
Since `len(waypoints)` is the number of waypoints, but python uses zero indexing, the largest valid index is `len(waypoints) - 1`. The goal here is to ensure that `(i+ 1)` is valid, so we need that `i + 1 <= len(waypoints) - 1` , or in other words, `i < len(waypoints) - 1`